### ISSUE TITLE
If need to re-set classification head (fc layer) at each new assignment?

### ISSUE THREAD

**Issue by haohang96** [yukimasano/self-label#8]
As I know,  DeepCluster need re-set classification head [https://github.com/facebookresearch/deepcluster/blob/9796a71abbfd14181a2b117d6244e60c2d94efbf/main.py#L145](url) because there is no mapping between two consecutive assignments. 

So in SeLa, if classification head is need to be re-set like deepcluster (It seems I can not find such code)? 

If re-set is not need, could you please explain why?  

Thanks!

**Comment by haohang96** [comment]
My simple intuition is in SeLa, the assignment Q is computed by logits P. So the prediction and assignment have some correlations and consistency. 

However, in deepcluster, the assignment is computed by features of last conv, so the logits and assignments is irrelevant.

But I don't konw if such explanation is reasonable.

**Comment by yukimasano** [comment]
Hi again, 

> If re-set is not need, could you please explain why?

because our assignments stay constant. It means an image that was cluster `k` previously is very likely to stay at cluster `k` and so on.

> My simple intuition is in SeLa, the assignment Q is computed by logits P. So the prediction and assignment have some correlations and consistency.

Correct.

> However, in deepcluster, the assignment is computed by features of last conv, so the logits and assignments is irrelevant.

The only reason you have to reset is because k-means is stochastic and the clusters could be permuted (e.g. what was the first cluster could now be the 3rd cluster and so on, this would massively confuse the fc layer).

Also please use the github issues page only for code related issues but feel free to contact me for any other questions.
